src/romi-config.ts

```diff
@@ -0,0 +1,80 @@
+import { DEFAULT_IO_CONFIGURATION, IOPinMode, IPinConfiguration, NUM_CONFIGURABLE_PINS } from "./romi-robot";
+import jsonfile from "jsonfile";
+import ProgramArguments from "./program-arguments";
+
+export interface RomiConfigJson {
+    ioConfig: string[]
+}
+
+export default class RomiConfiguration {
+    private _extIOConfig: IPinConfiguration[] = [];
+
+    constructor(programArgs: ProgramArguments) {
+        // Pre-load the external IO configuration
+        DEFAULT_IO_CONFIGURATION.forEach(val => this._extIOConfig.push(Object.assign({}, val)));
+
+        let isConfigError: boolean = false;
+
+        if (programArgs.config !== undefined) {
+            try {
+                const romiConfig: RomiConfigJson = jsonfile.readFileSync(programArgs.config);
+
+                if (romiConfig && romiConfig.ioConfig && (romiConfig.ioConfig instanceof Array)) {
+                    const portConfigs: string[] = romiConfig.ioConfig;
+                    if (portConfigs.length !== NUM_CONFIGURABLE_PINS) {
+                        isConfigError = true;
+                        throw new Error("[CONFIG] Invalid number of IO port configurations");
```

Comment:
It would be more descriptive to add the expected number of ports and the configured number of ports in the error message.